src/urllib3/_collections.py

```diff
@@ -45,54 +95,66 @@ class RecentlyUsedContainer(MutableMapping):
         ``dispose_func(value)`` is called.  Callback which will get called
     """
 
-    ContainerCls = OrderedDict
+    _container: "OrderedDict[_KT, _VT]"
+    _maxsize: int
+    dispose_func: Optional[Callable[[_VT], None]]
+    lock: RLock
 
-    def __init__(self, maxsize=10, dispose_func=None):
+    def __init__(
+        self, maxsize: int = 10, dispose_func: Optional[Callable[[_VT], None]] = None
+    ) -> None:
+        super().__init__()
         self._maxsize = maxsize
         self.dispose_func = dispose_func
-
-        self._container = self.ContainerCls()
+        self._container = OrderedDict()
         self.lock = RLock()
 
-    def __getitem__(self, key):
+    def __getitem__(self, key: _KT) -> _VT:
         # Re-insert the item, moving it to the end of the eviction line.
         with self.lock:
             item = self._container.pop(key)
             self._container[key] = item
             return item
 
-    def __setitem__(self, key, value):
-        evicted_value = _Null
+    def __setitem__(self, key: _KT, value: _VT) -> None:
+        evicted_value = None
         with self.lock:
             # Possibly evict the existing value of 'key'
-            evicted_value = self._container.get(key, _Null)
+            if key in self._container:
```

Comment:
Is the below code safe to do? Saves a lookup of `key in container`

```python
try:
    evicted_value = self._container.pop(key)
except KeyError:
    if len(self._container) >= self._maxsize:
        _, evicted_value = self._container.popitem(last=False)

self._container[key] = value
```

Reply:
Will take a peek.